### PR TITLE
File Transform - JSON variable substitution - Support all types

### PR DESCRIPTION
**Pull request by vincentdass**
#9257

### CODE REVIEW

File: Tasks/FileTransformV1/package-lock.json [hunk 1]
```diff
@@ -292,9 +292,9 @@
       "integrity": "sha512-cQKh8igo5QUhZ7lg38DYWAxMvjSAKG0A8wGSVimP07SIUEK2UO+arSRKbRZWtelMtN5V0Hkwh5ryOto/SshYIg=="
     },
     "ltx": {
-      "version": "2.6.2",
-      "resolved": "http://registry.npmjs.org/ltx/-/ltx-2.6.2.tgz",
-      "integrity": "sha1-cD5EN9XjlNJsAxT9j9Xevtjgmdk=",
+      "version": "2.8.0",
```
Comment (vincentdass) [inline Tasks/FileTransformV1/package-lock.json]: verified this version. 
approved and IP scanned.

**Comment by SumiranAgg** [comment]
Can we add value getting substituted in logs. Both for JsonVariableSubstitution and jsonVariableSubstitutionv2

**Comment by SumiranAgg** [comment]
Also great job adding L1s 👍

**Comment by vincentdass** [comment]
@SumiranAgg . thanks. Added debug statements.

**Comment by rposener** [comment]
So excited for this - going to solve several deployment challenges for me.  On pins and needles waiting for @kmkumaran to approve and see this get merged and be available!

**Comment by nikhil2007rock** [comment]
Can we get any info/doc about the syntax of substituting the whole JSON subgroup?
Like, I have the below in the config.json file
```
{
 "appDetails" : {
                 "www.contoso.com" : "12fbdc54-4ff7-48bc-bffa-35bd3ffcd2cf"
                }
}
```
I need to substitute both the key (www.contoso.com) and the value (12fbdc54-4ff7-48bc-bffa-35bd3ffcd2cf) under appDetails.

**Comment by SumiranAgg** [comment]
You can add a variable in your pipeline with key "appDetails" and value in json format lile {"www.abc.com": "someguid"}

This is the reference doc for the task https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/utility/file-transform?view=azure-devops.
We are still working on adding more examples and use cases here.

**Pull request merged**